.github/workflows/dev-branch-pr-build-and-test-pipeline.yaml

```diff
@@ -0,0 +1,35 @@
+name: Dev PR Build and Test Pipeline
+
+on:
+  pull_request:
+    branches:
+      - development
+
+jobs:
+  sonarqube:
+    name: Tests and SonarQube
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install dependencies
+        run: yarn
+
+      - name: Test and coverage
+        run: yarn jest --coverage
+
+      - name: SonarQube Analysis
+        uses: sonarsource/sonarqube-scan-action@master
+        env:
+          SONAR_TOKEN: sqp_d5037397d43452eb548ffb026abd6ecccbfce756
+          SONAR_HOST_URL: http://31.132.209.218:63535
+
+      - name: SonarQube Quality Gate check
+        uses: sonarsource/sonarqube-quality-gate-action@master
+        timeout-minutes: 5
+        env:
+           SONAR_TOKEN: sqp_d5037397d43452eb548ffb026abd6ecccbfce756
+           SONAR_HOST_URL: http://31.132.209.218:63535
```

Comment:
Should be replaced with `${{ secrets.SONAR_TOKEN }}` and `${{ secrets.SONAR_HOST_URL }}`